samcli/lib/build/workflow_config.py

```diff
@@ -97,6 +106,11 @@ def get_workflow_config(runtime, code_dir, project_dir):
     namedtuple(Capability)
         namedtuple that represents the Builder Workflow Config
     """
+    specified_build_workflow = metadata.get("BuildMethod", None) if metadata else None
```

Comment:
I understand why you are using `BuildMethod` to avoid conflict with existing runtime. However, I though the initial idea in [this](https://github.com/awslabs/aws-lambda-builders/pull/166/files#diff-cf7e02516f5f279dc897b4a2243e2d3e) design doc was to simply check if `makefile` exist with `build-{logical-id}` then use it or fallback to current runtime. This might confuse users.

Reply:
fair call out, the doc needs to be updated. But the change is not backwards incompatible and can be covered in docs to showcase its usage.